components/brave_federated/test/BUILD.gn

```diff
@@ -0,0 +1,27 @@
+# Copyright (c) 2021 The Brave Authors. All rights reserved.
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this file,
+# You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import("//brave/build/config.gni")
+import("//testing/test.gni")
+
+source_set("brave_ad_notification_timing_data_store_unit_tests") {
+  testonly = true
+  sources = [
+    "//brave/components/brave_federated/data_stores/ad_notification_timing_data_store_unittest.cc",
```

Comment:
in all `BUILD.gn`s you can use relative paths, no need for full ones

Reply:
also there is no need to have separate build.gn and separate subdir `test` for tests. One build.gn can contain everything